src/services/lease-service/priority-list-service.ts

```diff
@@ -0,0 +1,178 @@
+/*
+ * Service for:
+ * Fetching and consolidating detailed information on applicants from internal database and xpand
+ * Sorting applicants based on rental rules
+ */
+
+import {
+  Applicant,
+  Lease,
+  ParkingSpaceApplicationCategory,
+  parkingSpaceApplicationCategoryTranslation,
+  WaitingList,
+} from 'onecore-types'
+import { getWaitingList } from './adapters/xpand/xpand-soap-adapter'
+import {
+  getContactByContactCode,
+  getResidentialAreaByRentalPropertyId,
+  getLeasesForContactCode,
+} from './adapters/xpand/tenant-lease-adapter'
+import app from '../../app'
+
+const getDetailedApplicantInformation = async (applicant: Applicant) => {
+  try {
+    const applicantFromXpand = await getContactByContactCode(
+      applicant.contactCode,
+      'false'
+    )
+
+    if (!applicantFromXpand) {
+      throw new Error(
+        `Applicant ${applicant.contactCode} not found in contact query`
+      )
+    }
+
+    const applicantWaitingList = await getWaitingList(
+      applicantFromXpand.nationalRegistrationNumber
+    )
+    const waitingListForInternalParkingSpace =
+      parseWaitingListForInternalParkingSpace(applicantWaitingList)
+
+    if (!waitingListForInternalParkingSpace) {
+      throw new Error(
+        `Waiting list for internal parking space not found for applicant ${applicant.contactCode}`
+      )
+    }
+
+    const leases = await getLeasesForContactCode(
+      applicant.contactCode,
+      'true', //this filter does not consider upcoming leases
+      undefined //do not include contacts
+    )
+
+    if (!leases) {
+      throw new Error(`Leases not found for applicant ${applicant.contactCode}`)
+    }
+
+    let activeAndUpcomingLeases: Lease[] = leases.filter(
```

Comment:
```suggestion
    const activeAndUpcomingLeases: Lease[] = leases.filter(
```

Detta är också TS-error på min burk.. 🤔